xskillscore/core/np_deterministic.py

```diff
@@ -30,17 +32,18 @@ def _pearson_r(a, b, axis):
     """
     a = np.rollaxis(a, axis)
     b = np.rollaxis(b, axis)
-    ma = np.mean(a, axis=0)
-    mb = np.mean(b, axis=0)
+    weights = np.rollaxis(weights, axis)
+    ma = np.sum(a * weights, axis=0) / np.sum(weights, axis=0)
```

Comment:
This is problematic. I discovered that if you introduce a spatial mask with no weighting, this will create an array of all 1's spanning the full grid (ignoring the mask) and then generates a large denominator sum which drastically underestimates the value being computed depending on the mask region size.

I don't think the solution is to try to infer a mask, since users could have a NaN mask, zero mask, -999, etc. Instead there should be a fork created somewhere here to have *no* weights applied when `weights=None` is called, effectively recreating the current master branch of `xskillscore`. I have to think about how to do this smoothly so I don't bloat the code and double the line count here.